scripts/g.extension/testsuite/test_addons_download.py

```diff
@@ -0,0 +1,164 @@
+            files = os.listdir(self.install_prefix)
+            if files:
+                RuntimeError(
+                    "Install prefix path '{}' contains files {}".format(
+                        self.install_prefix, files
+                    )
+                )
+
+    def tearDown(self):
+        """Remove created files"""
+        silent_rmtree(self.install_prefix)
+
+    def test_github_install(self):
+        """Test installing extension from github"""
+
+        self.assertModule(
+            "g.extension",
+            extension="r.example.plus",
+            url="https://github.com/wenzeslaus/r.example.plus",
+            prefix=self.install_prefix,
+        )
+
+        # Modules with non-standard branch would be good for testing...
+        self.assertModule(
+            "g.extension",
+            extension="r.example.plus",
+            url="https://github.com/wenzeslaus/r.example.plus",
+            prefix=self.install_prefix,
+        )
+
+        for file in self.files:
+            self.assertFileExists(file)
+
+    def test_gitlab_install(self):
+        """Test installing extension from gitlab"""
+        self.assertModule(
+            "g.extension",
+            extension="r.example.plus",
+            url="https://gitlab.com/vpetras/r.example.plus",
+            prefix=self.install_prefix,
+        )
+
+        for file in self.files:
+            self.assertFileExists(file)
+
+    def test_bitbucket_install(self):
+        """Test installing extension from bitbucket"""
+        d_rast_files = [
+            os.path.join(self.install_prefix, "scripts", "r.sim.stats"),
+            os.path.join(self.install_prefix, "docs", "html", "r.sim.stats.html"),
+        ]
+        self.assertModule(
+            "g.extension",
+            extension="r.sim.stats",
+            url="https://bitbucket.org/lrntct/r.sim.stats",
+            prefix=self.install_prefix,
+        )
+
+        for file in d_rast_files:
+            self.assertFileExists(file)
+
+    def test_github_install_official(self):
+        """Test installing C-extension from official addons repository"""
+        r_gdd_files = [
+            os.path.join(self.install_prefix, "bin", "r.gdd"),
+            os.path.join(self.install_prefix, "docs", "html", "r.gdd.html"),
+        ]
+        self.assertModule("g.extension", extension="r.gdd", prefix=self.install_prefix)
+
+        for file in r_gdd_files:
+            self.assertFileExists(file)
+
+    def test_github_install_official_multimodule(self):
+        """Test installing multi-module extension from official addons repository"""
+        i_sentinel_files = [
+            os.path.join(
+                self.install_prefix, "scripts", "i.sentinel.parallel.download"
+            ),
+            os.path.join(
+                self.install_prefix, "docs", "html", "i.sentinel.parallel.download.html"
+            ),
+            os.path.join(self.install_prefix, "scripts", "i.sentinel.import"),
```

Comment:
self.install_prefix as Path may make this code much shorter. We are using it more and more in the code as is everybody else, so it should be quite readable.